src/client/Health/NodeView.jsx

```diff
@@ -25,37 +39,258 @@ const height = 500
+  sortableDataSources?.forEach((dataSource, index) => {
+    dataSource.index = index
+  })
+
+  const getDataSourceIndex = ((dataSourceId) => {
+    const position = sortableDataSources.findIndex((dataSource) => dataSource.dataSourceId === dataSourceId)
+    if (position === -1) {
+      return position
+    }
+    return sortableDataSources[position].index
+  })
+
+  const addDataSource = ({ dataSourceId, protocol }) => {
+    const dataSourceIndex = getDataSourceIndex(dataSourceId)
+    if (dataSourceIndex === -1) {
+      // Adds new dataSource
+      dispatchNewConfig({
+        type: 'addRow',
+        name: 'south.dataSources',
+        value: { dataSourceId, protocol, enabled: false },
+      })
+    } else {
+      const error = new Error('data source already exists')
+      setAlert({ text: error.message, type: 'danger' })
+      throw error
+    }
+  }
+
+  const handleDeleteSouth = (position) => {
+    dispatchNewConfig({ type: 'deleteRow', name: `south.dataSources.${sortableDataSources[position].index}` })
+  }
+
+  const handleDuplicateSouth = (position) => {
+    const dataSource = dataSources[sortableDataSources[position].index]
+    const newName = `${dataSource.dataSourceId} copy`
+    const countCopies = dataSources.filter((e) => e.dataSourceId.startsWith(newName)).length
+    dispatchNewConfig({
+      type: 'addRow',
+      name: 'south.dataSources',
+      value: {
+        ...dataSource,
+        dataSourceId: `${newName}${countCopies > 0 ? countCopies + 1 : ''}`,
+        enabled: false,
+      },
+    })
+  }
+
+  const handleStatusSouth = (position) => {
+    const dataSource = sortableDataSources[position]
+    const pathname = `/south/${dataSource.dataSourceId}/live`
+    history.push({ pathname })
+  }
+
+  const handleEditSouth = (position) => {
+    const dataSource = sortableDataSources[position]
+    const pathname = `/south/${dataSource.dataSourceId}`
+    history.push({ pathname })
+  }
+
+  const handleDataSourceIdChanged = (dataSourceIndex, newDataSourceId) => {
+    setRenamingConnector(null)
+    dispatchNewConfig({
+      type: 'update',
+      name: `south.dataSources.${dataSourceIndex}.dataSourceId`,
+      value: newDataSourceId,
+    })
+  }
+  // End Tools *******************************
 
   const northNodes = applications?.map((application, index) => (
+
     {
       id: application.applicationId, // unique id of node
       type: 'output', // output node
       targetPosition: 'bottom', // handle is at the bottom
       style: {
-        background: colors.background[application.enabled ? 'enabled' : 'disabled'],
-        border: colors.border[application.enabled ? 'enabled' : 'disabled'],
-        width: '140px',
+        background: colors.background[application.enabled ? 'disabled' : 'disabled'],
+        border: colors.border[application.enabled ? 'disabled' : 'disabled'],
+        width: '170px',
+        height: '25%',
       },
```

Comment:
Can you pass a css class to the node ? It would be better since you reuse the same style for the south. If you cannot pass a css class, just pass a standard object instead of rewriting the same code for the south